plugins/orchestrator-backend/src/service/DevModeService.ts

```diff
@@ -0,0 +1,235 @@
+    }
+
+    launcherArgs.push('--rm');
+    launcherArgs.push('-e', `QUARKUS_HTTP_PORT=${this.connection.port}`);
+
+    launcherArgs.push('-p', `${this.connection.port}:${this.connection.port}`);
+    launcherArgs.push('-e', `KOGITO_SERVICE_URL=${this.devModeUrl}`);
+    launcherArgs.push(
+      '-v',
+      `${resourcesAbsPath}:${SONATA_FLOW_RESOURCES_PATH}`,
+    );
+    launcherArgs.push('-e', 'KOGITO.CODEGEN.PROCESS.FAILONERROR=false');
+    launcherArgs.push(
+      '-e',
+      `QUARKUS_EMBEDDED_POSTGRESQL_DATA_DIR=${this.connection.persistencePath}`,
+    );
+
+    if (this.connection.jira) {
+      launcherArgs.push(
+        '-e',
+        'QUARKUS_REST_CLIENT_JIRA_OPENAPI_JSON_URL=http://jira.test:8080 -e ',
+      );
+      launcherArgs.push(`JIRABEARERTOKEN=${this.connection.jira.bearerToken}`);
+    }
+
+    launcherArgs.push(this.connection.containerImage);
+
+    return {
+      command: 'docker',
+      args: launcherArgs,
+    };
+  }
+
+  private extractConnectionConfig(config: Config): DevModeConnectionConfig {
+    const host =
+      config.getOptionalString('orchestrator.sonataFlowService.baseUrl') ??
+      DEFAULT_SONATAFLOW_BASE_URL;
+    const port = config.getOptionalNumber(
+      'orchestrator.sonataFlowService.port',
+    );
+
+    const resourcesPath =
+      config.getOptionalString(
+        'orchestrator.sonataFlowService.workflowsSource.localPath',
+      ) ?? '';
+
+    const containerImage =
+      config.getOptionalString('orchestrator.sonataFlowService.container') ??
+      DEFAULT_SONATAFLOW_CONTAINER_IMAGE;
+
+    const persistencePath =
+      config.getOptionalString(
+        'orchestrator.sonataFlowService.persistence.path',
+      ) ?? DEFAULT_SONATAFLOW_PERSISTANCE_PATH;
+
+    const jiraHost = config.getOptionalString('orchestrator.jira.host');
+    const jiraBearerToken = config.getOptionalString(
+      'orchestrator.jira.bearerToken',
+    );
+
+    const jiraConfig: JiraConfig | undefined =
+      jiraHost && jiraBearerToken
+        ? {
+            host: jiraHost,
+            bearerToken: jiraBearerToken,
+          }
+        : undefined;
+
+    const repoUrl =
+      config.getOptionalString(
+        'orchestrator.sonataFlowService.workflowsSource.gitRepositoryUrl',
+      ) ?? '';
+
+    return {
+      host,
+      port,
+      containerImage,
+      resourcesPath,
+      persistencePath,
+      jira: jiraConfig,
+      repoUrl,
+    };
+  }
+
+  public async loadDevWorkflows() {
+    if (!this.connection.repoUrl) {
+      this.logger.info('No Git repository configured. Skipping reload.');
+      return;
+    }
+
+    this.logger.info(`Reloading workflows from ${this.connection.repoUrl}`);
+    const localPath = this.connection.resourcesPath;
+    if (await fs.pathExists(localPath)) {
+      this.logger.info(`Path ${localPath} already exists. Skipping clone.`);
+      return;
+    }
+
+    await fs.remove(localPath);
```

Comment:
why do we need this `remove`?
in line 227 you check if `pathExists`, so in line 232 there is no need to remove.

Reply:
Good catch. This code was only moved from `SonataFlowService` but I think we can remove this line.